server/schedule/hbstream/heartbeat_streams.go

```diff
@@ -40,34 +40,46 @@ type streamUpdate struct {
 	stream  opt.HeartbeatStream
 }
 
-type heartbeatStreams struct {
+type HeartbeatStreams struct {
 	wg             sync.WaitGroup
 	hbStreamCtx    context.Context
 	hbStreamCancel context.CancelFunc
 	clusterID      uint64
 	streams        map[uint64]opt.HeartbeatStream
 	msgCh          chan *pdpb.RegionHeartbeatResponse
 	streamCh       chan streamUpdate
-	cluster        *cluster.RaftCluster
+	storeInformer  core.StoreSetInformer
+	noNeedRun      bool // For test only.
 }
 
-func newHeartbeatStreams(ctx context.Context, clusterID uint64, cluster *cluster.RaftCluster) *heartbeatStreams {
+func NewHeartbeatStreams(ctx context.Context, clusterID uint64, storeInformer core.StoreSetInformer) *HeartbeatStreams {
+	return newHeartbeatStreams(ctx, clusterID, storeInformer, false)
+}
+
+func NewTestHeartbeatStreams(ctx context.Context, clusterID uint64, storeInformer core.StoreSetInformer, noNeedRun bool) *HeartbeatStreams {
+	return newHeartbeatStreams(ctx, clusterID, storeInformer, noNeedRun)
+}
+
+func newHeartbeatStreams(ctx context.Context, clusterID uint64, storeInformer core.StoreSetInformer, noNeedRun bool) *HeartbeatStreams {
```

Comment:
consider change `noNeedRun` to `needRun`? Reversed boolean is not easy to understand.

Reply:
I just want to keep it the same as before. 

But I think you are right.